kfs-purap/src/main/java/edu/arizona/kfs/module/purap/service/impl/PurapAccountingServiceImpl.java

```diff
@@ -0,0 +1,147 @@
+import org.kuali.kfs.module.purap.PurapConstants;
+import org.kuali.kfs.module.purap.businessobject.PurApAccountingLine;
+import org.kuali.kfs.module.purap.businessobject.PurApItem;
+import org.kuali.kfs.module.purap.businessobject.PurApItemUseTax;
+import org.kuali.kfs.module.purap.document.PurchasingAccountsPayableDocument;
+import org.kuali.kfs.module.purap.util.UseTaxContainer;
+import org.kuali.kfs.sys.businessobject.AccountingLineBase;
+import org.kuali.kfs.sys.businessobject.SourceAccountingLine;
+import org.kuali.kfs.sys.util.ObjectPopulationUtils;
+import org.kuali.rice.core.api.util.type.KualiDecimal;
+
+import edu.arizona.kfs.module.purap.service.PurapAccountingService;
+
+
+public class PurapAccountingServiceImpl extends org.kuali.kfs.module.purap.service.impl.PurapAccountingServiceImpl 
+   implements PurapAccountingService {
+    private static org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(PurapAccountingServiceImpl.class);
+    
+    @Override
+    public List<PurApAccountingLine> generateAccountDistributionForProration(List<SourceAccountingLine> accounts, KualiDecimal totalAmount, Integer percentScale, Class clazz, List<PurApItem> purApItems) {
+        String methodName = "generateAccountDistributionForProration()";
+        if ( LOG.isDebugEnabled() ) {
+            LOG.debug(methodName + " started");
+        }
+                
+        if(totalAmount.isZero() || isAllSummaryAccoountsAmountZero(accounts)) {
+            return getFirstItemAccountsForDistribution(purApItems, clazz);
+        } else {
+            return super.generateAccountDistributionForProration(accounts, totalAmount, percentScale, clazz);
+        }
+    }
+
+    /*
+     * Check if all the summary accounts amount are zero. If all are zero, it will cause exception when creating
+     * distribution account.
+     */
+    private boolean isAllSummaryAccoountsAmountZero(List<SourceAccountingLine> summaryAccounts) {
+    	boolean isZero = true;
+    	for (SourceAccountingLine account : summaryAccounts) {
+    		isZero = account.getAmount().isZero();
+    		if (!isZero) {
+    			break;
+    		}
+    	}
+    	return isZero;
+    	
+    }
+
+    /*
+     * get first line item's accounts as distribution accounts
+     */
+    private List<PurApAccountingLine> getFirstItemAccountsForDistribution(List<PurApItem> purapItems, Class clazz) {
+        List<PurApAccountingLine> newAccounts = new ArrayList<PurApAccountingLine>();
+    	
+        for (PurApItem item : purapItems) {
+            if (item.getItemType().isLineItemIndicator()) {
+            	for (PurApAccountingLine account : item.getSourceAccountingLines()) {
+                    PurApAccountingLine newAccountingLine = null;
+                    try {
+                        newAccountingLine = (PurApAccountingLine) clazz.newInstance();
+                    }
+                    catch (InstantiationException e) {
+                        e.printStackTrace();
```

Comment:
I'm not a fan of printing STEs directly to STDOUT -- I'd suggest either log out at an appropriate level, or swallow the exception entirely with a comment explaining why that's not a bad thing (e.g.: // ObjectPopulationUtils can handle a this nulled out reference... I have no idea if that's the case).